### ISSUE TITLE
alternative assembler and running SDA locally

### ISSUE THREAD

**Issue by dcopetti** [mrvollger/SDA#14]
Hi,
I am wondering whether to run SDA on my plant assembly (>4 Gb, about 5 Gb expected size) where I have collapsed regions (they are about 400 Mb I think). I obtained it from about 40x ONT data (N50 18 kb).
The best assembly was generated by Flye, as Canu would have taken too long, wtdbg2 gave a very collapsed (2 Gb, ctg N50 40 kb) assembly. I am wondering whether it is possible to have another assembly running in there (Flye would be the best I think), or if we can run the assemblies outside of the SDA pipeline.
I have tried reassembling independently such regions with Canu already, and gained only 70 Mb more.
Also, I have access to a 1 TB RAM, 96 CPU, 20 TB storage machine: is this a good resource to run SDA or will it just do better on a cluster?
Thanks,

Dario

PS: any idea of running time on my machine? like 2 days vs 2 months or so

**Comment by mrvollger** [comment]
Hi Dario,

You should be able to run `SDA denovo` locally on a machine with those specs without issue in about 2 days. Though I have not run plant genomes before. 
```
SDA denovo --platform ont --species {rm_db} --threads 96 --fofn {fofn} --ref {flye.fasta} -n
```
`{rm_db}`: name of a repeat masker database for you species.
`{fofn}`: File of paths to you raw ONT data.
`{flye.fasta}`: Path to your flye assembly.
(remove the `-n` to execute instead of just a dry run).


As far as altering SDA so that it attempts local Flye assemblies as well as Canu and wtdbg2 I am afraid I don't have the bandwidth to do in the short term. However `sda.phased.readids` (one of the outputs) gives the read id and paralog group for each different SDA collapse. I don't think it would be too hard to parse this output, make different groups of reads, and then perform flye assemblies. 

Here is an example of the top of that file:
```
prefix    readid                                                                           cut  unique
region_0  m160514_174025_00116_c101018322550000001823229610211687_s1_p0/129182/0_31109     0    True
region_0  m160629_005250_00116_c101017112550000001823229610211660_s1_p0/97124/34742_40631  0    True
region_0  m141203_053752_00116_c100742522550000001823143306051587_s1_p0/44948/0_15722      0    True
region_0  m160630_220858_00116_c100962252550000001823220307011636_s1_p0/35760/0_24938      0    True
region_0  m141215_064445_00116_c100753262550000001823161007081525_s1_p0/129501/0_4432      0    True
region_0  m160617_032710_42274_c100976702550000001823226708101674_s1_p0/82614/28365_38700  0    True
region_0  m141108_221101_00116_c100741642550000001823143306051546_s1_p0/25562/12396_22547  0    True
region_0  m160606_181057_00116_c101013912550000001823231410211674_s1_p0/114611/0_35769     0    True
region_0  m141127_165420_00116_c100741732550000001823143306051525_s1_p0/57206/16782_24780  0    True
region_1  m141227_135939_00116_c100680942550000001823133702221557_s1_p0/77239/3066_11165   2    True
region_1  m141107_201310_00116_c100741642550000001823143306051540_s1_p0/9015/0_19809       2    True
region_1  m160607_025447_00116_c101013912550000001823231410211676_s1_p0/39032/21361_27675  2    True
region_1  m160708_100122_00116_c101018602550000001823229610211615_s1_p0/124695/0_23602     2    True
region_1  m160703_004840_00116_c101017922550000001823229610211614_s1_p0/57641/34117_40055  2    True
region_1  m141202_080147_00116_c100742522550000001823143306051582_s1_p0/98085/2123_7061    2    True
region_1  m160704_215023_00116_c101018262550000001823229610211675_s1_p0/110950/0_19425     2    True
region_1  m141129_100106_00116_c100741762550000001823143306051596_s1_p0/60788/982_16094    2    True
region_1  m141213_222338_00116_c100755882550000001823162407081576_s1_p0/160242/0_3682      2    True
region_1  m160513_040713_42274_c101014032550000001823231410211611_s1_p0/153109/0_10981     2    True
```

The idea would be to get all the read ideas for a particular prefix and cut pair and then execute a flye assembly on those reads. And then repeat for all the prefix, cut pairs in the file.

I hope that helps!

Best,
Mitchell
